superset-frontend/src/components/Chart/DrillBy/DrillByModal.tsx

```diff
@@ -150,17 +178,23 @@ export default function DrillByModal({
         }
       `}
       show={showModal}
-      onHide={onHideModal ?? (() => null)}
+      onHide={() => {
+        setShowChart(true);
+        setChartDataResult(undefined);
```

Comment:
What do you think about conditionally rendering the modal in `DrillByMenuItems`? Like, instead of passing `showModal` prop to `DrillByModal`, we could do `{showModal && <DrillByModal ... />}`. That way, we probably wouldn't need to worry about cleaning up on modal hide.